game-service/src/main/java/org/libertybikes/game/round/service/GameRoundWebsocket.java

```diff
@@ -86,17 +86,27 @@ else if (!round.isPlayer(session))
                 session.close();
                 return;
             }
-            // log(roundId, "[onMessage] msg=" + message);
+            //log(roundId, "[onMessage] msg=" + message);
 
             if (GameEvent.GAME_START == msg.event) {
                 round.startGame();
             } else if (msg.direction != null) {
                 round.updatePlayerDirection(session, msg);
             } else if (msg.playerJoinedId != null) {
-                org.libertybikes.restclient.Player playerResponse = getPlayer(msg.playerJoinedId);
-                if (!round.addPlayer(session, msg.playerJoinedId, playerResponse.name, msg.hasGameBoard))
-                    sendToClient(session, new OutboundMessage.ErrorEvent("Unable to add player " + playerResponse.name
-                                                                         + " to game. This is probably because someone else with the same name is already in the game."));
+                // TODO: Handle a bot joining and returning null from player DB better.
+                if (!msg.playerJoinedId.startsWith("BOT:")) {
```

Comment:
do we need special handling here? IMO its fine to let Bot players be registered in the player DB.

Reply:
I think the frontend creates them because it's just trying to retrieve them right here and is getting NPE.

Reply:
we could either have the game-service lazily create a non-existing player that joins, or we could have the AI registration flow call player-service and register itself

Reply:
That's why it's a TODO :P

Reply:
ok that's fine -- the TODO comment said "handle bot joining ...." and my point was that the game service should not need to handle a bot players any differently than real players